### PR TITLE
RELATED: RAIL-3844 - Remove use of contexts to pass around props

### PR DESCRIPTION
**Pull request by lupko**
These are essential changes to enable/improve interop between dynamically loaded engine and locally embedded customizations. In that case, there may be multiple contexts coming locally and coming from the bundle and these are independent on each other. The problems start happening because the local code cannot retrieve props that are set in bundled engine's context.

Additionally, the PR delivers config to allow passing local dashboard's react-redux context where the bundled engine should save the redux store. This is needed for the useDashboardSelect/useDashboardDispatch to work in locally contributed components embedded into dynamically loaded dashboard.

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by lupko** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5833/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7597/ : SUCCESS in 18m 26s

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dateFilter.spec.ts**|7|-|13.3180s|
|✅|**drilling.spec.ts**|2|-|9.1420s|
|✅|**dashboard.spec.ts**|9|-|14.2840s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/65/).

File: tools/plugin-toolkit/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 build
 test-plugin*
+with-engine
```
Comment (no23reason) [inline tools/plugin-toolkit/.gitignore:3]: Probably a leftover?

Comment (lupko) [reply tools/plugin-toolkit/.gitignore]: indeed. removed

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1925

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1670/ : SUCCESS in 3m 12s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1950/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/747/ : SUCCESS in 16m 04s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7602/ : SUCCESS in 20m 40s

**Pull request merged**